apps/workspace-host/src/interface.ts

```diff
@@ -622,13 +655,19 @@ async function _pullImage(workspace) {
   // the first successful pull. This allows us to reference
   // the previously pulled layers and provide a more accurate
   // percantage calculation on any subsequent pulls.
-  let progressDetails = (await cache.get(`workspaceProgressInit:${workspace_image}`)) || {};
-  let progressDetailsInit = {};
+  const progressDetails: Record<
+    string,
+    {
+      current: number;
+      total: number;
+    }
+  > = (await cache.get(`workspaceProgressInit:${workspace_image}`)) || {};
```

Comment:
Nit: this feels like lying to the compiler, though obviously it's no worse that what we do now. If we're making this proper TypeScript, I'd rather run this though Zod and default to an empty object if it fails validation.